pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+for what is expected to be returned
+"""
+from __future__ import absolute_import, print_function
+
+import copy
+import uuid
+import time
+from mock import Mock
+import six
+
+from pywbem import WBEMConnection, CIMClass, CIMClassName, \
+    CIMInstance, CIMInstanceName, CIMQualifierDeclaration, \
+    CIMError, \
+    CIM_ERR_NOT_FOUND, CIM_ERR_FAILED, CIM_ERR_INVALID_SUPERCLASS, \
+    CIM_ERR_INVALID_PARAMETER, CIM_ERR_INVALID_CLASS, CIM_ERR_ALREADY_EXISTS, \
+    CIM_ERR_INVALID_NAMESPACE, CIM_ERR_INVALID_ENUMERATION_CONTEXT, \
+    CIM_ERR_NOT_SUPPORTED, \
+    DEFAULT_NAMESPACE, \
+    MOFCompiler, MOFWBEMConnection
+from pywbem.cim_obj import NocaseDict
+
+
+__all__ = ['FakedWBEMConnection']
+
+# This is a server default if the open request MaxObjectCount is None
+_DEFAULT_MAX_OBJECT_COUNT = 101
+
+
+class FakedWBEMConnection(WBEMConnection):
+    """
+    A faked WBEMConnection class that mocks _imethodcall and _methodcall
+    WBEMConnection methods and creates equivalents for all of
+    the WBEMConnection methods that communicate with the server.
+    Because it only mocks _imethodcall and _methodcall all of the logic in
+    the various WBEMConneciton methods that prepare requests for the server
+    (ex. GetClass) is utilized and the parameters of the method calls are
+    tested/modified, etc.
+    """
+    def __init__(self, verbose=None):
+        super(FakedWBEMConnection, self).__init__(
+            'http://FakedUrl', default_namespace=DEFAULT_NAMESPACE,
+            # creds=None,
+            # x509=None, verify_callback=None, ca_certs=None,
+            # no_verifi cation=False, timeout=None,
+            use_pull_operations=False,
+            enable_stats=False)
+
+        # TODO we are not useing NocaseDict.  Are namespace names case
+        # independent?????
+        # Dictionary of classes representing a repository.  This is
+        # a dictionary of dictionaries where the top level key is
+        # namespace and the keys for each dictionary in a namespace are
+        # class names in the repository. The values in each subdirectory are
+        # CIMClass objects.
+        self.classes = {}
+
+        # Same format as classes above, dictionary of dictionaries where
+        # the top level keys are namespaces and the next level kesy are
+        # CIMQualifierDeclaration names with values of the
+        # CIMQualifierDeclaration objects
+        self.qualifiers = {}
+
+        # Top level dictionary: key is namespace,
+        # and value is list of instances in that namespace
+        # TODO we should really have a subdict per class.
+        self.instances = {}
+
+        # Open Pull Contexts. The key for each context is an enumeration
+        # context id.  The data is the total list of instances/names to
+        # be returned and the current position in the list. Any context in
+        # this list is still open.
+        self.enumeration_contexts = {}
+
+        self.verbose = verbose
```

Comment:
The use of verbose is really pretty much for debug purposes. I don't think we need that. Instead, we should make sure the logging works, and that would be the way to trace calls and their returns.

Reply:
Agreed.  It got us through debugging but will move to logging what we output before release.

Agreed but code not cleaned up yet.

Reply:
I created TODO in code to move the prints to a logger.